src/ExecutorContainer.ts

```diff
@@ -68,27 +79,13 @@ export default class ExecutorContainer extends EventEmitter implements Iterable<
 	 * @returns a new executor associated with the given language and file
 	 */
 	private createExecutorFor(file: string, language: LanguageId, needsShell: boolean) {
-		if (this.plugin.settings[`${language}Interactive`] || this.isAlwaysInteractive(language)) {
-			switch (language) {
-				case "js":
-					return new NodeJSExecutor(this.plugin.settings, file);
-				case "python":
-					return new PythonExecutor(this.plugin.settings, file);
-				case "prolog":
-					return new PrologExecutor(this.plugin.settings, file);
-			}
-		}
+		// Interactive language executor
+		if (this.plugin.settings[`${language}Interactive`])
+			return new interactiveExecutors[language](this.plugin.settings, file);
+		// Custom non-interactive language executor
+		else if (Object.keys(nonInteractiveExecutors).contains(language))

```

Comment:
Maybe use `language in nonInteractiveExecutors` instead of `Object.keys(...).contains(...)` for readability & small performance boost. There's a *possible* bug if we ever add languages named `constructor`, `hasOwnProperty`, `toString`, `valueOf`, etc. but I think that is negligible in exchange for the readability.